pdr_backend/analytics/predictoor_dashboard/dash_components/plots.py

```diff
@@ -0,0 +1,151 @@
+import plotly.graph_objects as go
+from enforce_typing import enforce_types
+from pdr_backend.util.time_types import UnixTimeS
+
+
+@enforce_types
+def create_scatter(
+    name: str, x_data: List[float], y_data: List[int], mode="lines"
+) -> go.Scatter:
+    """Create a scatter plot with the given data.
+    Args:
+        name (str): Name of the trace.
+        x_data (List): List of x values.
+        y_data (List): List of y values.
+        mode (str): Plot mode. Default is 'lines'.
+    Returns:
+        go.Scatter: Scatter plot trace.
+    """
+    return go.Scatter(x=x_data, y=y_data, mode=mode, name=name)
+
+
+@enforce_types
+def create_bar(name: str, x_data: List[float], y_data: List[int]) -> go.Bar:
+    """Create a bar plot with the given data.
+    Args:
+        name (str): Name of the trace.
+        x_data (list): List of x values.
+        y_data (list): List of y values.
+    Returns:
+        go.Bar: Bar plot trace.
+    """
+    return go.Bar(x=x_data, y=y_data, name=name, width=1)
+
+
+@enforce_types
+def process_payouts(payouts: List[dict], predictor: str, feed: str) -> tuple:
+    """
+    Process payouts data for a given predictor and feed.
+    Args:
+        payouts (list): List of payouts data.
+        predictor (str): Predictor address.
+        feed (str): Feed contract address.
+    Returns:
+        tuple: Tuple of slots, accuracies, profits, and stakes.
+    """
+    slots, accuracies, profits, stakes = [], [], [], []
+    profit = predictions = correct_predictions = 0
+
+    for p in payouts:
+        if predictor in p["ID"] and feed in p["ID"]:
+            predictions += 1
+            profit_change = p["payout"] - p["stake"] if p["payout"] > 0 else -p["stake"]
+            profit += profit_change
+            correct_predictions += p["payout"] > 0
+
+            slots.append(p["slot"])
+            accuracies.append((correct_predictions / predictions) * 100)
+            profits.append(profit)
+            stakes.append(p["stake"])
+    slot_in_date_format = [
+        UnixTimeS(ts).to_milliseconds().to_dt().strftime("%m-%d %H:%M") for ts in slots
+    ]
+    return slot_in_date_format, accuracies, profits, stakes
+
+
+@enforce_types
+def create_figure(
+    data_traces: List[Union[go.Scatter, go.Bar]],
+    title: str,
+    yaxis_title: str,
+    show_legend: bool = True,
+):
+    """
+    Create a figure with the given data traces.
+    Args:
+        data_traces (list): List of data traces.
+        title (str): Figure title.
+        yaxis_title (str): Y-axis title.
+        show_legend (bool): Show legend. Default is True.
+    Returns:
+        go.Figure: Plotly figure.
+    """
+    fig = go.Figure(data_traces)
+    fig.update_layout(
+        title=title,
+        yaxis_title=yaxis_title,
+        margin={"l": 20, "r": 0, "t": 50, "b": 0},
+        showlegend=show_legend,
+        xaxis_nticks=4,
+        bargap=0.1,
+        barmode="stack",
+        legend=(
```

Comment:
I would move the legend part separately, to avoid confusion.
i.e. before fig.update layout